fdbserver/DataDistributionTracker.actor.cpp

```diff
@@ -813,12 +813,60 @@ ACTOR Future<Void> fetchShardMetrics( DataDistributionTracker* self, GetMetricsR
 	return Void();
 }
 
+
+ACTOR Future<Void> fetchShardMetricsList_impl( DataDistributionTracker* self, GetMetricsListRequest req ) {
+	try {
+		loop {
+			// used to control shard limit
+			int shardNum = 0;
+			// list of metrics, regenerate on loop when full range unsuccessful
+			Standalone<VectorRef<DDMetricsRef>> result;
+			Future<Void> onChange;
+			for( auto t : self->shards.intersectingRanges( req.keys ) ) {
```

Comment:
@tclinken  @sfc-gh-anoyes  This seems unclear to me. I may need your opinions.
If the `req.keys` are the interesting range, for example, we have `req.keys = ("b", "c")`. Then should we only return `bytes` of data-distribution from `b` to `c`? 
However, if the underlying data distribution is from `a` to `d`, then we do not know the exact bytes here. 
The `intersectingRanges` used here will give back the bytes from `a` to `d`. However, this is not compatible with special-key-space. In particular, if we call `getRange("\xff\xff/data_distribution/b", "\xff\xff/data_distribution/c")`, it will give back the result with one key-value pair: 
`"\xff\xff/data_distribution/a"`: `<bytes_from_a_to_d>`
Thus, I may need to change this one to `containedRanges`. In this way, if the underlying range is larger than your query range, nothing will be returned.

Reply:
I think if we have a shard `[a,d]` and request the data distribution metrics for `[b,c]` it makes sense to return the metrics for the whole `[a,d]` shard, instead of nothing. Maybe `getRange("\xff\xff/data_distribution/b", "\xff\xff/data_distribution/c")` could return `"\xff\xff/data_distribution/b": <bytes_from_a_to_d>`?

Reply:
I think this one works. If we do it in this way, the `bytes` here is like an upper bound for the query range. The user is not able to tell whether it is the number of the exact bytes. However, if this upper bound is good enough for the usage, then it is definitely okay to do so.

Reply:
I think we should just focus on mapping the information to kv pairs, and then respecting the semantics of the requested getrange. If the user wants the behavior @tclinken described above they can `getrange(lastLessOrEqual(b), c)`

Reply:
@tclinken I do not have a strong preference to return nothing in this case. But as @sfc-gh-anoyes said, we can use key selectors to achieve the same goal. In this case, returning nothing looks more consistent with the semantics. Do you have a use case where this may cause confusion?

Reply:
@sfc-gh-anoyes makes a good point, we can switch to `containedRanges`, and for the above query I think we'd want to use `getRange(lastLessOrEqual(LiteralStringRef("\xff\xff/data_distribution/b")), firstGreaterOrEqual(LiteralStringRef("\xff\xff/data_distribution/c")))`.

Reply:
Cool. Thanks for the clarification.